### PR TITLE
Feat/hfm 20 erc721 enforcer tests

### PR DESCRIPTION
**Pull request by djaciel**
## Describe Changes

- Migrate test from file `04_erc721_enforcer_tests_l1.ts` to Foundry test suite
- Test can be found in the file `04_Erc721Enforcer.t.sol`
- All test migrated, except for the test _"should fail initializing already initialized ERC721 Enforcer"_ due to I cannot found the sampleErc721Enforcer deployed contract. Marked as skipped
- Add _"All Foundry tests are passing"_ to the PR checklist template

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [ ] All Hardhat tests are passing

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -10,3 +10,4 @@ I made this more better by doing ...
 - [ ] I have performed a self-review of my code
 - [ ] Code styles have been enforced
 - [ ] All Hardhat tests are passing
+- [ ] All Foundry tests are passing
```
Comment (lucas-marc) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: praise: nice thinking!

File: test/foundry/deploy/04_Erc721Enforcer.t.sol [hunk 2]
```diff
@@ -0,0 +1,827 @@
...
+
+  string public constant tokenURI1 = "https://holograph.xyz/sample1.json";
+  string public constant tokenURI2 = "https://holograph.xyz/sample2.json";
+  string public constant tokenURI3 = "https://holograph.xyz/sample3.json";
+
+
+  /// @notice Set up the testing environment by initializing all necessary contracts and accounts.
+  function setUp() public {
+    uint256 localHostFork = vm.createFork(LOCALHOST_RPC_URL);
+    vm.selectFork(localHostFork);
```
Comment (lucas-marc) [inline test/foundry/deploy/04_Erc721Enforcer.t.sol]: suggestion: we can use createSelectFork here, to do both things on one line.

Comment (lucas-marc) [reply test/foundry/deploy/04_Erc721Enforcer.t.sol]: More info here: https://book.getfoundry.sh/cheatcodes/create-select-fork

File: test/foundry/deploy/04_Erc721Enforcer.t.sol [hunk 3]
```diff
@@ -0,0 +1,827 @@
...
+  /// @dev Initializes testing accounts.
+  function _setupAccounts() private {
+    deployer = vm.addr(0xff22437ccbedfffafa93a9f1da2e8c19c1711052799acf3b58ae5bebb5c6bd7b);
+    alice = vm.addr(1);
+    bob = vm.addr(2);
+    charlie = vm.addr(3);
+  }
+
+  /// @dev Deploys the contract instances used in the tests.
+  function _deployContracts() private {
```
Comment (lucas-marc) [inline test/foundry/deploy/04_Erc721Enforcer.t.sol]: suggestion: we are not actually deploying contracts here, I think _setupContracts might be a better naming, as it's correct for the current behaviour, and also works if we need to instantiate something in the future.

File: test/foundry/deploy/04_Erc721Enforcer.t.sol
Comment (lucas-marc) [inline test/foundry/deploy/04_Erc721Enforcer.t.sol:1]: suggestion: can we add natSpecs comments to all tests, to clarify what is being tested and state to which test on hardhat they relate to?

File: test/foundry/deploy/04_Erc721Enforcer.t.sol [hunk 4]
```diff
@@ -0,0 +1,827 @@
...
+  function _mint(address _to, uint256 _tokenId, string memory _tokenURI) private {
+    vm.prank(deployer);
+    sampleERC721.mint(_to, uint224(_tokenId), _tokenURI);
+  }
+
+  /*
+   * CHECK INTERFACES
+   */
+
+  function testUntilDeployBalanceOf() public {
```
Comment (lucas-marc) [inline test/foundry/deploy/04_Erc721Enforcer.t.sol]: suggestion: not sure why we are using this "UntilDeploy" prefix on all tests

**Review by lucas-marc — COMMENTED** [review]
Partial review, left a few comments

**Review by sogoiii — APPROVED** [review]
👍

**Pull request merged**